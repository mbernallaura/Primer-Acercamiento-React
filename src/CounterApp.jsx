import { useState } from 'react';
import PropTypes from 'prop-types';


export const CounterApp = ({value}) =>{
    
    //!Cuando el estado (useState) cambia, vulve y carga la funcion CounterApp
    const [counter, setCounter] = useState( value );

    const incremento = () => {
        //!De ambas maneras se puede hacer
        //setCounter((c) => c + 1 );
        setCounter( counter + 1);
    }

    const decremento = () => setCounter(counter - 1);
    

    const reset = () => setCounter(value);
    

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            <button onClick={ incremento }> +1 </button>
            <button onClick={ decremento }> -1 </button>
            <button aria-label='btn-reset' onClick={ reset }> Reset </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

