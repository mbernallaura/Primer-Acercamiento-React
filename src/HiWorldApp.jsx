
//!Componenentes React = Funciones
export const HiWorldApp = () =>{
  return (
    <h1>Hi World App</h1>
  )
}
