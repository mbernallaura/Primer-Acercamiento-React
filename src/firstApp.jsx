import PropTypes from 'prop-types';

const newMessage = [1,2,3,4,5,6,7,8,9];
const objeto = {
    message: 'Hola Mundo',
    title: 'Laura',
}

const Saludo = () =>{
    return `Hola ${objeto.title}`;
}

export const FirstApp = ({title, subTitle, name}) =>{
    //!Simpre hay que regresar al menos un Nodo Padre y adentro puede llevar varios elementos
    //!Si no se quiere regresar un div para enviar varios elementos se puede usar <> </>
    //!Para fragentar el codigo"

    //!Se puede devolver Funciones que NO SEAN ASYNCRONAS

    return (
        <> 
            <h1 data-testid= 'test-title'> { title } </h1>
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
            <p>{ name }</p>
            <p>{ Saludo() }</p>
            {/* <code>{ JSON.stringify( objeto ) }</code> */}
            <div>
                <ul>
                    <li>Nombre</li>
                    <li>Apellido</li>
                </ul>
            </div>
        </>
    )
}


//!Se utlizan los propTypes para hacer validaciones en las variables
//!Se debe instalar para que funcione (npm install prop-types)
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
}


//!Se puede crear nuevos proptypes por medio de default
FirstApp.defaultProps = {
    //title: 'No hay Titulo',
    subTitle: 'No hay subtitulo',
    name: 'Laura'
}