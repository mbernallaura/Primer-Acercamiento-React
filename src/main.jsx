import React from 'react';
import ReactDOM from 'react-dom/client';
//import { HiWorldApp } from './HiWorldApp';
import { FirstApp } from './firstApp';
import './styles.css';

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <FirstApp title='Hola, soy Goku' subTitle={123}/>
    </React.StrictMode>
);