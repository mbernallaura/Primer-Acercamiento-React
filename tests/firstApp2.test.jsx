import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/firstApp";

describe('Pruebas en <firstApp/>', () => {
    const title = 'Hola soy Goku';
    const subtitle = 'Soy un subtitulo';

    test('Debe de hacer match con el snapshot', () => {
        const {container} = render(<FirstApp title={ title }/>)

        //! toMatchSnapshot() = Hace una captura del componente y lo guarda enn un archivo que lo crea
        expect(container).toMatchSnapshot();
    });

    test('Debe mostar el mensaje "Hola soy Goku"', () => {
        render(<FirstApp title={ title }/>);
        expect(screen.getByText(title)).toBeTruthy();
        //screen.debug();
    });

    test('Debe de mostrar el titulo en un H1', () => {
        render(<FirstApp title={ title }/>);

        //! getByRole() = busca cuaquier etiqueta
        expect(screen.getByRole('heading', { level: 1}).innerHTML).toContain(title);
    })

    test('Debe de enviar el subtitulo por props', () => {
        render(
            <FirstApp 
                title= { title }
                subTitle= { subtitle }
            />
        );
        expect(screen.getAllByText(subtitle).length).toBe(2);
    })
});