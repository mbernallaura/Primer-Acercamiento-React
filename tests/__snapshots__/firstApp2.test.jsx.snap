// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <firstApp/> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy Goku
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Laura
  </p>
  <p>
    Hola Laura
  </p>
  <div>
    <ul>
      <li>
        Nombre
      </li>
      <li>
        Apellido
      </li>
    </ul>
  </div>
</div>
`;
